src/pages/Lending/Pool/components/Repay/RepayConfirm.tsx

```diff
@@ -73,17 +74,30 @@ export const RepayConfirm = ({
     accountId: collateralAccountId,
   })
 
+  const { mutateAsync } = useMutation({
+    mutationKey: [txId],
+    mutationFn: async (_txId: string) => {
+      // Enforcing outbound checks when repaying 100% since that will trigger a collateral refund transfer
+      // which we *want* to wait for before considering the repay as complete
+      await waitForThorchainUpdate({ txId: _txId, skipOutbound: bn(repaymentPercent).lt(100) })
+        .promise
+      queryClient.invalidateQueries({ queryKey: ['thorchainLendingPosition'], exact: false })
+    },
+  })
+
+  const lendingMutationStatus = useMutationState({
+    filters: { mutationKey: [txId] },
+    select: mutation => mutation.state.status,
+  })
+
+  const loanTxStatus = useMemo(() => lendingMutationStatus?.[0], [lendingMutationStatus])
   useEffect(() => {
     // don't start polling until we have a tx
-    if (!txHash) return
-
-    setIsLoanClosePending(true)
+    if (!txId) return
     ;(async () => {
-      await waitForThorchainUpdate({ txHash, queryClient, skipOutbound: true }).promise
-      setIsLoanClosePending(false)
-      await refetchLendingPositionData()
+      await mutateAsync(txId)
     })()
```

Comment:
github wont let me add a suggstion here, but you can do this
```typescript
;mutateAsync(txId)
```

instead of 
```typescript
    ;(async () => {
      await mutateAsync(txId)
    })()
```